### PR TITLE
Remove security emailer

### PR DESCRIPTION
**Pull request by jdillon**
Based on https://github.com/sonatype/nexus-oss/pull/970

Removes the emailing of things from security, which is pretty poop

### CODE REVIEW

**Comment by jdillon** [comment]
http://bamboo.s/browse/NX-OSSF502-4 :white_check_mark:

**Comment by mcculls** [comment]
+1

**Comment by adreghiciu** [comment]
+1

**Comment by cstamas** [comment]
+1

File: components/nexus-security/src/main/java/org/sonatype/security/DefaultSecuritySystem.java [hunk 1]
```diff
@@ -664,101 +618,11 @@ public void changePassword(String userId, String newPassword)
     eventBus.post(new UserPrincipalsExpired(userId, user.getSource()));
   }
 
-  public void forgotPassword(String userId, String email)
```
Comment (kellyrob99) [inline components/nexus-security/src/main/java/org/sonatype/security/DefaultSecuritySystem.java]: Is there any plan to replace this functionality? Forgotten password handling is seemingly the only valuable "feature" we're losing with this removal.

Comment (jdillon) [inline components/nexus-security/src/main/java/org/sonatype/security/DefaultSecuritySystem.java]: This feature was already removed from the UI (and thus the product), this was only removing the underpinnings.

https://issues.sonatype.org/browse/NEXUS-7294 is to replace this with something sane and secure.

Comment (kellyrob99) [inline components/nexus-security/src/main/java/org/sonatype/security/DefaultSecuritySystem.java]: tyvm @jdillon

**Pull request merged**